gufe/tests/test_models.py

```diff
@@ -41,23 +69,135 @@ def test_default_settings():
+            with pytest.raises(ValueError):
+                _ = OpenMMSystemGeneratorFFSettings(nonbonded_cutoff=value)
+
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            ("pme", True, "pme"),
+            ("NOCUTOFF", True, "NOCUTOFF"),
+            ("no cutoff", False, None),
+            (1.0, False, None),
+        ],
+    )
+    def test_openmmff_nonbonded_method(self, value, valid, expected):
+        if valid:
+            s = OpenMMSystemGeneratorFFSettings(nonbonded_method=value)
+            assert s.nonbonded_method == expected
+        else:
+            with pytest.raises(ValueError, match="Only PME and NoCutoff are allowed"):
+                _ = OpenMMSystemGeneratorFFSettings(nonbonded_method=value)
+
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            (298 * unit.kelvin, True, 298 * unit.kelvin),
+            (298, True, 298 * unit.kelvin),
+            (298.0, True, 298 * unit.kelvin),
+            ("298 kelvin", True, 298 * unit.kelvin),
+            ("298", False, None),
+            (298 * unit.angstrom, False, None),
+        ],
+    )
+    def test_thermo_temperature(self, value, valid, expected):
+        if valid:
+            s = ThermoSettings(temperature=value)
+            assert s.temperature == expected
+        else:
+            with pytest.raises(ValueError):
+                _ = ThermoSettings(temperature=value)
+
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            (1.0 * unit.atm, True, 1.0 * unit.atm),
+            (1.0, True, 1.0 * unit.atm),
+            ("1 atm", True, 1.0 * unit.atm),
+            ("1.0", False, None),
+        ],
+    )
+    def test_thermo_pressure(self, value, valid, expected):
+        if valid:
+            s = ThermoSettings(pressure=value)
+            assert s.pressure == expected
+        else:
+            with pytest.raises(ValueError):
+                _ = ThermoSettings(pressure=value)
+
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            (1.0, True, 1.0),
+            (1, True, 1.0),
+            ("1 ph", False, None),
+            (1.0 * unit.atm, False, None),
+        ],
+    )
+    def test_thermo_ph(self, value, valid, expected):
+        if valid:
+            s = ThermoSettings(ph=value)
+            assert s.ph == expected
+        else:
+            with pytest.raises(ValueError):
+                _ = ThermoSettings(ph=value)
+
+    @pytest.mark.parametrize(
+        "value,valid,expected",
+        [
+            (1.0, True, 1.0),
+            (None, True, None),
+            ("1", True, 1.0),
+        ],
+    )
+    def test_thermo_redox(self, value, valid, expected):
+        if valid:
+            s = ThermoSettings(redox_potential=value)
+            assert s.redox_potential == expected
+        else:
+            with pytest.raises(ValueError):
+                _ = ThermoSettings(redox_potential=value)
```

Comment:
In a future PR, we should think about how we can re-factor this pattern into some testing method where we pass in the class name and setting as well so it will be easy to add more without having to copy this bit